content/posts/vim-command-line-window/index.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Vim's missing feature isn't really missing
+author: Qasim Warraich
+comments: true
+date: 2023-12-29
+---
+
+Since it's near the end of 2023, I am sure everyone is quite busy with family
+and other festive obligations. I am going keep this one short and introduce you
+to one Vim feature that, if you don't use it already, will have you excited to
+get back to your beloved editor in 2024.
+
+<!--more-->
+
+## An Addiction
+
+![`:w` everywhere](./colon-w.png)
+
+If you are anything like me, then Vim is so deeply ingrained in you that you find
+yourself accidentally typing `:w` or its friends in all sorts of text inputs.
+Vim motions have taken over your brain to the point that you find it difficult
+to type in “traditional” text inputs. Perhaps your Vim addiction is so bad that
+you have Vi mode configured everywhere you can.
+
+You feel like a well-oiled machine, navigating your browser, your file system,
+and even editing commands in your shell using Vim motions. However, when you
+hit `:` to enter your beloved Vim's `command` mode and want to follow up with
+something more than `w` or `q`, something feels off. You make a typo or forget
+a flag, and there it is. Your fingers seek awkwardly for the arrow keys, you're
+reaching for your backspace, and every time you inadvertently hit escape, it's
+game over.
+
+## Command Line Window
+
+The so-called, [Command-line
+window](https://neovim.io/doc/user/cmdline.html#cmdline-window) (`:h
+command-line-window`) is one of the less talked-about Vim features, and one that
+I personally discovered sometime ago by pure chance.
+
+It is a special window in Vim that allows you to edit text for `command-line`
+operations in the same way you edit text normally in Vim. This allows you to
+circumvent the awkwardness of searching for arrow keys or using backspace,
+allowing you to use the Vim motions you are so comfortable with in even more
+contexts. Additionally, the `command-line window` also populates the buffer
+with your command or search history from previous operations, allowing you to
+modify a recently run command or search using Vim motions. Another advantage of
+using `command-line window` is that you can also hook into your non-native
+autocompletion suggestions for `command-line` operations.
+
+The command-line window can be accessed in several ways:
+
+- `q:` : Opens `command-line window` populated with your recent `Ex` commands
+- `q/` or `q?` : Opens `command-line window` populated with your recent
+  searches. Hitting enter executes a search using the `/` or `?` behaviour.
+- `CTRL+f` : Brings up the `command-line-window` when you are already in
+  `Command-line mode`.
```

Comment:
check for the instances of <some> mode and be consistent in highlighting and capitalization.

Reply:
Yes, also found this a bit strange but modes are always capitalised in the vim help docs.